### ISSUE TITLE
Too many types provided for mongoose.Model

### ISSUE THREAD

**Issue by EnigmaticHamster** [Bounced-Inc/mongoose-tsgen#19]
Privided interface:
```javascript
export interface StoreModel
extends mongoose.Model<StoreDocument, StoreQueries>,
    StoreStatics {}
```
Gives the error:
interface Model<T extends mongoose.Document>
Generic type 'Model<T>' requires 1 type argument(s).ts(2314)

**Comment by francescov1** [comment]
Are the other referenced interfaces present as well? Please provide a full example of Mongoose schema and generated interface

**Comment by francescov1** [comment]
I see your error, seems to be related to the new official Mongoose typings https://github.com/Automattic/mongoose/issues/8108. I'll look to add support over the weekend, for now you can use this with any version of Mongoose below v5.11.

**Comment by francescov1** [comment]
Seems like the [`QueryHelpers` parameter on the Model interface in @types/mongoose](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mongoose/index.d.ts#L3193) has been removed [in the official Mongoose types. ](https://github.com/Automattic/mongoose/blob/master/index.d.ts#L590). Poked around the official Mongoose types but wasn't able to find a new way to provide the `QueryHelper` type for a Model (these currently allow us to properly type `Schema.query` functions on the Model).

Will likely open an issue with Mongoose about it after a bit more searching. If you know a way to do this let me know!

**Comment by www-chique** [comment]
Not sure if it's related, but according to the doc,
```
UserSchema.methods = {
  isMetadataString() {
    //code here
  }
} as UserMethods;
```
isn't working.

Instead, have to use
```
UserSchema.methods.isMetadataString = function() {
     //code here
}
```
And it perfectly works!

It's same for statics, and I believe query as well.

I'm a big fan of **mongoose-tsgen**, if somehow there is a possible fix to this problem, it'll be super amazing.

**Comment by francescov1** [comment]
Hey @www-chique, 
Mongoose made a breaking change to their type definitions with the release of v5.11, if you downgrade mongoose versions it should fix the issue (see https://github.com/Bounced-Inc/mongoose-tsgen/issues/19#issuecomment-747901114 above).

To support the new Mongoose version, need to find a way to create a Model type with custom query definitions (with the new type definitions, they've removed the second generic argument for `Model` which was previously used for passing query types). Pretty busy for the next week or so but will give it a go after that, any help is appreciated if you have some time!

**Comment by luishrg** [comment]
Just waiting for de fix to use the package on my project. I'm amazed by this package, even if a can't use it

**Comment by francescov1** [comment]
Hey friends just wanted to provide a quick update on this, I'm chatting with someone involved in the Mongoose project and it seems the query typing that we're trying to achieve here is not currently possible with the official types.

Apparently because of the immaturity of the new official types, the guy is [maintaining his own fork](https://github.com/alttypes/mongoose) for more complete types. I'm going to check it out and see what the differences are, but it likely makes sense to try and stick with the official mongoose types if possible so that we don't need users to install separate typings. 

For now it may make sense to check a user's mongoose version and handle typing differently for v5.11 and above. This would mean we'd likely just remove the query typing for that subset of users, but hopefully can add it back once its supported. The guy mentioned he might know a few potential workarounds for getting the query types so I'll explore those as well.

**Comment by francescov1** [comment]
Support for Mongoose 5.11+ released in version 7.0.0 🚀 

For the query typing, I had to augment mongoose's `Query` type but was able to attain the same level of type safety as before. Let me know if you run into any unsupported edge cases, there's a lot of small changes with the new Mongoose types so could be a few extra things that need to be handled.

**Comment by www-chique** [comment]
Amazing. Will definitely update and try it out and report back if I find something.

Many thanks for your efforts. I can't go back to typing Interfaces again.
